src/parser/statement.rs

```diff
@@ -42,6 +42,13 @@ pub fn statement(tokens: TokenList) -> ParseResult<Statement> {
         }
     }
 
+    // Statement can end if the last token is an empty statement.
+    if let Some(last_item) = next_tokens.next() {
```

Comment:
Nit but this should be "statement can end if the _next_ token is an empty statement" (and I guess then also `next_item` below that). The branch above says "last token" since it looks backwards in the token list.